src/workerd/jsg/util.c++

```diff
@@ -570,29 +571,60 @@ kj::Array<kj::byte> asBytes(v8::Local<v8::ArrayBufferView> arrayBufferView) {
 }
 
 void recursivelyFreeze(v8::Local<v8::Context> context, v8::Local<v8::Value> value) {
-  if (value->IsArray()) {
-    // Optimize array freezing (Array is a subclass of Object, but we can iterate it faster).
-    v8::HandleScope scope(v8::Isolate::GetCurrent());
-    auto arr = value.As<v8::Array>();
-
-    for (auto i: kj::zeroTo(arr->Length())) {
-      recursivelyFreeze(context, check(arr->Get(context, i)));
-    }
+  auto isolate = v8::Isolate::GetCurrent();
+  v8::HandleScope outerScope(isolate);
+  kj::Vector<v8::Local<v8::Value>> queue;
+  queue.reserve(128);
+  queue.add(value);
+
+  while (!queue.empty()) {
+    auto item = queue.back();
+    queue.removeLast();
+
+    if (item->IsArray()) {
```

Comment:
An Array can also have named properties, do we care? Looks like the original version ignored those.

Reply:
I think the same argument for circular references applies here too. If we're getting this from a parsed json structure then no named properties will matter. A comment in here along those lines is helpful for future generations.